openunmix/filtering.py

```diff
@@ -6,6 +6,19 @@
 import torch.nn as nn
 from torch import Tensor
 from torch.utils.data import DataLoader
+import numpy as np
+
+
+def my_atan2(y, x):
+    pi = torch.from_numpy(np.array([np.pi])).to(y.device, y.dtype)
+    ans = torch.atan(y / x)
+    ans += ((y > 0) & (x < 0)) * pi
+    ans -= ((y < 0) & (x < 0)) * pi
+    ans *= (1 - ((y > 0) & (x == 0)) * 1.0)
+    ans += ((y > 0) & (x == 0)) * (pi / 2)
+    ans *= (1 - ((y < 0) & (x == 0)) * 1.0)
+    ans += ((y < 0) & (x == 0)) * (-pi / 2)
+    return ans
```

Comment:
please test what happens with `x==0`

Reply:
plus, be careful about in-place transformations `+=`: not sure they're ok for letting the gradient flow

Reply:
Ok, It is corrected